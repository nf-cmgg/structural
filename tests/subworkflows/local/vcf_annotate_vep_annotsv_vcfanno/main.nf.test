nextflow_workflow {

    name "Test Workflow VCF_ANNOTATE_VEP_ANNOTSV_VCFANNO"
    script "subworkflows/local/vcf_annotate_vep_annotsv_vcfanno/main.nf"
    workflow "VCF_ANNOTATE_VEP_ANNOTSV_VCFANNO"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_annotate_vep_annotsv_vcfanno"

    options "-stub"
    config "./nextflow.config"

    test("homo_sapiens - sv, no_filter") {

        when {
            params {
                annotations_filter = null
                genome = "GRCh38"
                species = "homo_sapiens"
                vep_cache_version = 110
                annotsv_file_name = "annotsv_annotated"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male", variant_type:"sv", family: "test1", family_count:1],
                    file(params.sv_vcf1, checkIfExists:true),
                    file(params.sv_tbi1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"test", sample:"test", sex:"male", family: "test1", family_count:1],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = []
                input[8] = []
                input[9] = []
                input[10] = []
                input[11] = ["sv"]
                input[12] = params.genome
                input[13] = params.species
                input[14] = params.vep_cache_version
                input[15] = null
                input[16] = []
                input[17] = files("\${params.vcfanno_tomls}/*.toml", checkIfExists:true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs[0][1..2].collect { file(it).name },
                    workflow.out.reports,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - cnv, filter") {

        when {
            params {
                annotations_filter = "-i FILTER='PASS'"
                genome = "GRCh38"
                species = "homo_sapiens"
                vep_cache_version = 110
                annotsv_file_name = "annotsv_annotated"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male", variant_type:"cnv", family: "test1", family_count:1],
                    file(params.sv_vcf1, checkIfExists:true),
                    file(params.sv_tbi1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"test", sample:"test", sex:"male", family: "test1", family_count:1],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = []
                input[8] = []
                input[9] = []
                input[10] = []
                input[11] = ["cnv"]
                input[12] = params.genome
                input[13] = params.species
                input[14] = params.vep_cache_version
                input[15] = params.annotations_filter
                input[16] = []
                input[17] = files("\${params.vcfanno_tomls}/*.toml", checkIfExists:true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs[0][1..2].collect { file(it).name },
                    workflow.out.reports,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - sv & cnv") {

        when {
            params {
                annotations_filter = null
                genome = "GRCh38"
                species = "homo_sapiens"
                vep_cache_version = 110
                annotsv_file_name = "annotsv_annotated"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male", variant_type:"sv", family: "test1", family_count:1],
                    file(params.sv_vcf1, checkIfExists:true),
                    file(params.sv_tbi1, checkIfExists:true)
                ], [
                    [id:"test", sample:"test", sex:"male", variant_type:"cnv", family: "test1", family_count:1],
                    file(params.cnv_vcf1, checkIfExists:true),
                    file(params.cnv_tbi1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"test", sample:"test", sex:"male", family: "test1", family_count:1],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = []
                input[8] = []
                input[9] = []
                input[10] = []
                input[11] = ["sv", "cnv"]
                input[12] = params.genome
                input[13] = params.species
                input[14] = params.vep_cache_version
                input[15] = null
                input[16] = []
                input[17] = files("\${params.vcfanno_tomls}/*.toml", checkIfExists:true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs[0][1..2].collect { file(it).name },
                    workflow.out.reports,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }
}
