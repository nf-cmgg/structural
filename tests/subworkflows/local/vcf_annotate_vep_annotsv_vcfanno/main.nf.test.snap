{
    "homo_sapiens - sv, no_filter": {
        "content": [
            [
                "test.sv.vcf.gz",
                "test.sv.vcf.gz.tbi"
            ],
            [
                "test.vep.summary.html:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            22,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T10:16:55.291447426"
    },
    "homo_sapiens - sv & cnv": {
        "content": [
            [
                "test.cnv.vcf.gz",
                "test.cnv.vcf.gz.tbi"
            ],
            [
                "test.vep.summary.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.vep.summary.html:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            44,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T10:17:27.514868326"
    },
    "homo_sapiens - cnv, filter": {
        "content": [
            [
                "test.cnv.vcf.gz",
                "test.cnv.vcf.gz.tbi"
            ],
            [
                "test.vep.summary.html:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            23,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:04:12.980338516"
    }
}